python/servo/testing_commands.py

```diff
@@ -146,13 +149,35 @@ def test_unit(self, test_name=None, package=None):
         self.ensure_bootstrapped()
 
         if package:
-            packages = [package]
+            packages = {package}
         else:
-            packages = os.listdir(path.join(self.context.topdir, "tests", "unit"))
+            packages = set()
+
+        test_patterns = []
+        for test in test_name:
+            # add package if 'tests/unit/<package>'
+            match = re.search("tests/unit/(\\w+)/?$", test)
```

Comment:
FYI, there's an [`fnmatch`](https://docs.python.org/2/library/fnmatch.html) library built into Python that would probably be more appropriate for matching filenames, but this is simple enough that it doesn't really matter